include/misc/__assert.h

```diff
@@ -92,14 +92,25 @@ extern void posix_exit(int exit_code);
 	}
 #endif
 
+#define __ASSERT_LOC(test)                               \
+	(void)printk("ASSERTION FAIL [%s] @ %s:%d\n",    \
+	       _STRINGIFY(test),                         \
+	       __FILE__,                                 \
+	       __LINE__)                                 \
+
+#define __ASSERT_NO_MSG(test)                                            \
+	do {                                                             \
+		if (!(test)) {                                           \
+			__ASSERT_LOC(test);                              \
+			__ASSERT_POST;                                   \
+		}                                                        \
+	} while (false)
+
 #define __ASSERT(test, fmt, ...)                                         \
 	do {                                                             \
 		if (!(test)) {                                           \
-			(void)printk("ASSERTION FAIL [%s] @ %s:%d:\n\t", \
-			       _STRINGIFY(test),                         \
-			       __FILE__,                                 \
-			       __LINE__);                                \
-			(void)printk(fmt, ##__VA_ARGS__);                \
+			__ASSERT_LOC(test);                              \
+			(void)printk("\t" fmt "\n", ##__VA_ARGS__);      \
```

Comment:
what's the purpose of `(void)` if printk does not return value?

Reply:
So, not me puts it there, not me removes. And MISRA is always around the corner...

Reply:
See a1135620ba8a82294e12fdd601f1ac98aa065b58

And there're a lot, a lot of cleanup to do in zephyr codebase, but not everything in one commit of course...

Reply:
i see. Nobody expects MISRA